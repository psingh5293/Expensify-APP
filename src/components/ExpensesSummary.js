import React from 'react';
import { connect } from 'react-redux';
import totalExpenses from '../selectors/expenses-total';
import expenses from '../selectors/expenses'
import numeral from 'numeral'


export class ExpensesSummary extends React.Component{


    render(){
        const expenseWord = this.props.expensesCount===1 ? 'expense' :'expenses';
        const formattedTotal = numeral(this.props.expensesTotal/100).format('$0,0.00')
        return(
            <div>
          <p> {`viewing ${this.props.expensesCount} ${expenseWord} totalling ${formattedTotal}`}</p>
            </div>
        );
    }

};

const mapStateToProps =(state)=> {
    const visibleExpenses = expenses(state.expenses,state.filters)
    return {
        expensesCount : visibleExpenses.length,
        expensesTotal : totalExpenses(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)