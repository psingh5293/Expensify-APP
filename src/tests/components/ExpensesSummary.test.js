import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import {shallow} from 'enzyme';

import expenses from '../fixtures/expenses'
import {filters} from '../fixtures/filters'
console.log(filters)
test('should render expenseSummary with one  expense',()=> {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={123} />)
    expect(wrapper).toMatchSnapshot();
})

