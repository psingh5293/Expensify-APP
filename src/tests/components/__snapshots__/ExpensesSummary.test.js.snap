// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseSummary with one  expense 1`] = `
<div>
  <p>
     
    viewing 1 expense totalling $1.23
  </p>
</div>
`;
